### PR TITLE
Петров Олег. Задача 1. Вариант 25. Технология SEQ. Линейная фильтрация изображений (вертикальное разбиение). Ядро Гаусса 3x3.

### PR DESCRIPTION
**Pull request by ValeraDanger**

### CODE REVIEW

File: tasks/seq/petrov_o_vertical_image_filtration/func_tests/main.cpp [hunk 1]
```diff
@@ -0,0 +1,140 @@
+#include <gtest/gtest.h>
+
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+#include "core/task/include/task.hpp"
+#include "seq/petrov_o_vertical_image_filtration/include/ops_seq.hpp"
+
```
Comment (aobolensk) [inline tasks/seq/petrov_o_vertical_image_filtration/func_tests/main.cpp:10]: Please, add random tests here as well

Comment (aobolensk) [reply tasks/seq/petrov_o_vertical_image_filtration/func_tests/main.cpp:10]: Also add smaller images tests (1x1, empty image,...) and non-square images

**Review by aobolensk — CHANGES_REQUESTED** [review]

**Review by mateusxap — APPROVED** [review]
This code is good! I can’t find any problems

File: tasks/seq/petrov_o_vertical_image_filtration/src/ops_seq.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#include "seq/petrov_o_vertical_image_filtration/include/ops_seq.hpp"
+
+#include <cmath>
+#include <cstddef>
+// #include <iostream>
```
Comment (Applejack2004) [inline tasks/seq/petrov_o_vertical_image_filtration/src/ops_seq.cpp:5]: delete comments

File: tasks/seq/petrov_o_vertical_image_filtration/include/ops_seq.hpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ public:
+  explicit TaskSequential(ppc::core::TaskDataPtr task_data) : Task(std::move(task_data)) {}
+  bool PreProcessingImpl() override;
+  bool ValidationImpl() override;
+  bool RunImpl() override;
+  bool PostProcessingImpl() override;
+
+ private:
+  std::vector<int> input_, output_;
+  // size_t rc_size_{};
```
Comment (Applejack2004) [inline tasks/seq/petrov_o_vertical_image_filtration/include/ops_seq.hpp:21]: delete comments

**Review by Applejack2004 — CHANGES_REQUESTED** [review]

File: tasks/seq/petrov_o_vertical_image_filtration/src/ops_seq.cpp (hunk 2 above)
Comment (ValeraDanger) [reply tasks/seq/petrov_o_vertical_image_filtration/src/ops_seq.cpp]: Done

File: tasks/seq/petrov_o_vertical_image_filtration/include/ops_seq.hpp (hunk 3 above)
Comment (ValeraDanger) [reply tasks/seq/petrov_o_vertical_image_filtration/include/ops_seq.hpp]: Done

**Review by Applejack2004 — APPROVED** [review]

File: tasks/seq/petrov_o_vertical_image_filtration/func_tests/main.cpp (hunk 1 above)
Comment (ValeraDanger) [reply tasks/seq/petrov_o_vertical_image_filtration/func_tests/main.cpp:11]: done

**Review by aobolensk — APPROVED** [review]

File: tasks/seq/petrov_o_vertical_image_filtration/func_tests/main.cpp [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+  EXPECT_EQ(out, expected_out);
+}
+
+TEST(petrov_o_vertical_image_filtration_seq, test_gaussian_filter_negative_data) {
+  constexpr size_t kWidth = 5;
+  constexpr size_t kHeight = 5;
+
+  // Create data
+  std::vector<int> in = {-1,  -2,  -3,  -4,  -5,  -6,  -7,  -8,  -9,  -10, -11, -12, -13,
+                         -14, -15, -16, -17, -18, -19, -20, -21, -22, -23, -24, -25};
```
Comment (allnes) [inline tasks/seq/petrov_o_vertical_image_filtration/func_tests/main.cpp:138]: how pixels can have negative values?

File: tasks/seq/petrov_o_vertical_image_filtration/func_tests/main.cpp [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+  test_task_sequential.PreProcessing();
+  test_task_sequential.Run();
+  test_task_sequential.PostProcessing();
+
+  // Check result
+  EXPECT_EQ(out, expected_out);
+}
+
+// TEST(petrov_o_vertical_image_filtration_seq, test_gaussian_filter_negative_data) {
+//   constexpr size_t kWidth = 5;
```
[the remark this answers is not in the document]
Comment (ValeraDanger) [reply tasks/seq/petrov_o_vertical_image_filtration/func_tests/main.cpp]: Done

**Review by sk1er52 — APPROVED** [review]

File: tasks/seq/petrov_o_vertical_image_filtration/func_tests/main.cpp (hunk 4 above)
Comment (ValeraDanger) [reply tasks/seq/petrov_o_vertical_image_filtration/func_tests/main.cpp]: removed

**Review by allnes — APPROVED** [review]
